sdgx/data_models/inspectors/numeric.py

```diff
@@ -14,69 +14,132 @@ class NumericInspector(Inspector):
 
     This class is a subclass of `Inspector` and is designed to provide methods for inspecting
     and analyzing numeric data. It includes methods for detecting int or float data type.
+
+    In August 2024, we introduced a new feature that will continue to judge the positivity or
+    negativity after determining the type, thereby effectively improving the quality of synthetic
+    data in subsequent processing.
+    """
+
+    int_columns: set = set()
+    """
+    A set of column names that contain integer values.
+    """
+
+    float_columns: set = set()
+    """
+    A set of column names that contain float values.
+    """
+
+    positive_columns: set = set()
+    """
+    A set of column names that contain only positive numeric values.
+    """
+
+    negative_columns: set = set()
+    """
+    A set of column names that contain only negative numeric values.
+    """
+
+    pos_threshold: float = 0.95
+    """
+    The threshold proportion of positive values in a column to consider it as a positive column.
+    """
+
+    negative_threshold: float = 0.95
+    """
+    The threshold proportion of negative values in a column to consider it as a negative column.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.int_columns: set[str] = set()
-        self.float_columns: set[str] = set()
         self._int_rate = 0.9
         self.df_length = 0
 
     def _is_int_column(self, col_series: pd.Series):
```

Comment:
More return type hint would be better:

`def _is_int_column(self, col_series: pd.Series) -> bool:`